From 7c0066ec7df9f5bfe4e79ca15ab5a36a279cecd8 Mon Sep 17 00:00:00 2001
From: Maital Hahn <maitalm@ti.com>
Date: Wed, 20 Apr 2016 16:21:03 +0300
Subject: [PATCH 37/47] mesh on demand: Perform BG Scan only on serving channel

The purpose of the BG scan in MoD, is to monitor the AP's RSSI.
Therefore, we can scan only the serving channel.

Signed-off-by: Maital Hahn <maitalm@ti.com>
---
 wpa_supplicant/bgscan_learn.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/wpa_supplicant/bgscan_learn.c b/wpa_supplicant/bgscan_learn.c
index cb732f7..c7f5208 100644
--- a/wpa_supplicant/bgscan_learn.c
+++ b/wpa_supplicant/bgscan_learn.c
@@ -416,7 +416,17 @@ static void * bgscan_learn_init(struct wpa_supplicant *wpa_s,
 			   "signal strength monitoring");
 	}
 
-	data->supp_freqs = bgscan_learn_get_supp_freqs(wpa_s);
+	if (wpa_s->global->mesh_on_demand.enabled)
+	{
+		data->supp_freqs = os_malloc(2*sizeof(int));
+		data->supp_freqs[0] = wpa_s->assoc_freq;
+		data->supp_freqs[1] = 0;
+	}
+	else
+	{
+		data->supp_freqs = bgscan_learn_get_supp_freqs(wpa_s);
+	}
+
 	data->scan_interval = data->short_interval;
 	if (data->signal_threshold) {
 		/* Poll for signal info to set initial scan interval */
-- 
1.9.1

